### PR TITLE
Revamp admin merchant dashboard

### PR DESCRIPTION
**Pull request by netia1128**
##Changes
-Adds model and features tests for the admin merchant user stories
-Moves merchant top_day from SQL into AR
-Other minor miscellaneous updates

##Checklist
 -Tests written where required
 -All tests passing

### CODE REVIEW

File: app/models/merchant.rb [hunk 1]
```diff
@@ -8,53 +8,68 @@ class Merchant < ApplicationRecord
...
-    select('merchants.name, merchants.id as merchant_id, sum(invoice_items.quantity * invoice_items.unit_price) as total_revenue').joins(:transactions).where("transactions.result = 'success'").group('merchants.id').order('total_revenue desc').limit(5)
+    select('merchants.name, merchants.id as id, sum(invoice_items.quantity * invoice_items.unit_price) as total_revenue')
+    .joins(:transactions)
+    .where("transactions.result = 'success'")
+    .group('merchants.id')
+    .order('total_revenue desc')
+    .limit(5)
   end
 
   def top_revenue_day
```
Comment (suzkiee) [inline app/models/merchant.rb:19]: Omg. This is a phenomenal refactor. 🙌

File: app/controllers/merchants/items_controller.rb [hunk 2]
```diff
@@ -1,6 +1,5 @@
 class Merchants::ItemsController < ApplicationController
   def index
-    require 'pry'; binding.pry
```
Comment (jamogriff) [inline app/controllers/merchants/items_controller.rb:3]: Thanks for fixing this. I just ran into that one this morning

File: spec/features/admin/merchants/index_spec.rb [hunk 3]
```diff
@@ -1,12 +1,100 @@
...
+      end
+    end
+    it 'tells you the top five merchants, their revenue, and best day' do
+      visit '/admin/merchants'
+
+      expect(page).to have_content('Top 5 Merchants')
+
+      within "#top-five-merchants > tr:nth-child(2)" do
+        expect(page).to have_content('Terry-Moore')
+        expect(page).to have_content('$3,115,616.00')
```
Comment (jamogriff) [inline spec/features/admin/merchants/index_spec.rb:38]: Where these numbers converted from cents into dollars?

**Review by jamogriff — APPROVED** [review]
Tests all day

**Pull request merged**